instruqt/support-cli/03-collect-support-bundle/assignment.md

```diff
@@ -0,0 +1,178 @@
+
+click on the *click here* link to show the commands to collect a support bundle via the CLI.
+
+![supportcli-kotsadm-supbundle3](../assets/supportcli-kotsadm-supbundle3.png)
+
+This is useful as sometimes the kotsadm UI may be unavailable and diagnosis needs to be performed from the CLI only.
+
+Note that the commands are in two parts.  Firstly install the support-bundle kubernetes cli plugin using krew:
+```
+curl https://krew.sh/support-bundle | bash
+```
+
+Note: If the kubernetes cluster was kURL then the support-bundle plugin would have been pre-installed.
+
+Then the bundle generation command:
+```
+kubectl support-bundle secret/default/kotsadm-support-cli-${INSTRUQT_PARTICIPANT_ID}-supportbundle --redactors=configmap/default/kotsadm-redact-spec/redact-spec,configmap/default/kotsadm-support-cli-${INSTRUQT_PARTICIPANT_ID}-redact-spec/redact-spec
+```
+
+Run both of the above commands now to generate the CLI support bundle.
+Note that the support-bundle cli command launches a console ui to view the bundle contents
+
+
+### 3. CLI Support Bundle Analysis
+
+Note the help output from the support-bundle:
+
+```
+kubectl support-bundle --help
+```
+
+![supportcli-supbundle-help](../assets/supportcli-supbundle-help.png)
+
+In the support bundle console UI, step through the issues and address them one by one.
+
+Here is some sample output similar to what you should see in your environment:
+![supportcli-supbundle1](../assets/supportcli-supbundle1.png)
+![supportcli-supbundle4](../assets/supportcli-supbundle4.png)
+
+
+### 4. Diagnose issues and fix application
+
+Based on the advice presented in the support bundle content, perform corrective action on the cluster..
+
+<details>
+  <summary>Open for a hint on config file issue</summary>
+
+Scrolling to the failing check, we can review the error message:
+![supportcli-supbundle2](../assets/supportcli-supbundle2.png)
+
+Specifically, you'll see the error message:
+```shell
+Could not find a file at /etc/support/config.txt with 400 permissions
+```
+
+To fix this, run:
+```shell
+chmod 400 /etc/support/config.txt
+```
+
+</details>
+
+<details>
+  <summary>Open for a hint on the restraining-bolt issue</summary>
+
+Scrolling to the failing check, we can review the error message:
+![supportcli-supbundle3](../assets/supportcli-supbundle3.png)
+
+Specifically, you'll see the error message:
+```shell
+Restraining bolt in /etc/support has short circuited the startup process. If you remove it, we might be able to launch the application.
+```
+
+We can remove this file with
+
+```shell
+rm /etc/support/restraining-bolt.txt
+```
+
+</details>
+
+
+### 5. App Status Check - CLI
+
+Check the status of the application services via the cli
+
+```
+kubectl get all
+```
+
+The nginx pod should be in *Running* state, it can take some time for it to loop after it has backed off so it can be deleted and the replica set will replace with a new one.
```

Comment:
```suggestion
The nginx pod should be in *Running* state, it can take some time for it to loop after it has backed off so it can be deleted and the replicaset will replace with a new one.
```